bodhi/server/consumers/masher.py

```diff
@@ -197,9 +200,19 @@ def work(self, msg):
                     if request == req:
                         self.log.info('Starting thread for %s %s for %d updates',
                                       release, request, len(updates))
-                        thread = MasherThread(release, request, updates, agent,
-                                              self.log, self.db_factory,
-                                              self.mash_dir, resume)
+                        if "modular" not in release:
```

Comment:
I'd rather not use the release's name to encode whether it should be modular or not. There are other Bodhi deployments in the wild and I can't guarantee they won't have Releases with "modular" in their names. We are receiving a list of updates to mash, however, and they have Builds and Builds have types.

Instead, we could launch both threads on every release and have each thread filter the given Updates for ones with ModularBuilds or RpmBuilds, respectively, and then proceed to mash only the content types they care about.

Reply:
You got this one, @bowlofeggs?

Reply:
This one needs to be done, and is up for grabs.

Reply:
@ralphbean @bowlofeggs 
So i was looking into this. A modular release should have only modular builds. So we could check the content type of some build of a release an choose according to this. The only problem is that the Masher class does have not access to the db. We could add it and then pass it to the MasherThread class. Or we could do what @bowlofeggs suggested yesterday that start the MasherThread and PungiMasherThread at the same time and the class will choose if the release has modular builds and continue or quit. Im not sure here. WDYT?